src/Microsoft.AspNet.Mvc.ModelBinding/ModelBindingContext.cs

```diff
@@ -119,6 +144,18 @@ public Type ModelType
         }
 
         /// <summary>
+        /// Gets or sets a model name which is explicitly set using an <see cref="IModelNameProvider"/>. 
+        /// <see cref="Model"/>.
+        /// </summary>
+        public string BinderModelName { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value which represents the <see cref="BindingSource"/> associated with the 
+        /// <see cref="Model"/>.
+        /// </summary>
+        public BindingSource BindingSource { get; set; }
+
+        /// <summary>
```

Comment:
Yeah, that needs to be done. I wanted the first cut to gear towards cleanup of the ModelMetadataProvider, I will update with the model binder changes.